src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -87,6 +98,27 @@ public void setMetadataConfig(SnowflakeMetadataConfig metadataConfigIn) {
     metadataConfig = metadataConfigIn;
   }
 
+  /**
+   * extract enableSchematization from the connector config and set the value for the recordService
+   *
+   * <p>The extracted boolean is returned for external usage.
+   *
+   * @param connectorConfig the connector config map
+   * @return a boolean indicating whether schematization is enabled
+   */
+  public boolean setEnableSchematizationFromConfig(final Map<String, String> connectorConfig) {
+    if (connectorConfig.containsKey(SnowflakeSinkConnectorConfig.SCHEMATIZATION_ENABLE_CONFIG)) {
+      enableSchematization =
+          Boolean.parseBoolean(
+              connectorConfig.get(SnowflakeSinkConnectorConfig.SCHEMATIZATION_ENABLE_CONFIG));
+    }
+    return enableSchematization;
+  }
+
+  public void setEnableSchematization(final boolean enableSchematizationIn) {
```

Comment:
annotate this with @ VisibleForTesting

Reply:
and javadoc which would entail this is only for testing. (Annotation does mean that so you can javadoc, but javadoc for @param would be great)